alacritty_terminal/src/term/cell.rs

```diff
@@ -92,6 +93,20 @@ impl Cell {
         self.extra.get_or_insert_with(Default::default).zerowidth.push(c);
     }
 
+    #[inline]
+    pub fn underline_color(&self) -> Option<Color> {
+        self.extra.as_ref().and_then(|extra| extra.underline_color)
+    }
```

Comment:
I'm not sure how much sense it makes to add a separate method for every single field in `extra`. Generally accessing this data is likely very performance sensitive, so small things can make big differences.

Having a single `pub fn extra(&self) -> Option<&CellExtra>` might make sense, because at that point the consumer of that method can probably do the exact same things as before but now the check for `None` on `extra` only needs to happen once, not multiple times.

Reply:
That could certainly be done. How would you imagine mutating the extra data? There could be a `pub fn extra_mut(&mut self) -> Option<&mut CellExtra>`, but that can only mutate existing extra data. A `pub fn set_extra(&mut self, extra: CellExtra)` would do unnecessary moves if there is already data there. Maybe we should then have both? Or `pub fn ensure_extra(&mut self)` which does the allocation?

Reply:
Currently I don't think there's any necessity to change tho way the data is mutated. The `push_zerowidth` function is in a different codepath that wouldn't have access to the mutable borrow of `extra_mut` anyways. So having two separate functions which both have to initialize it if it doesn't exist yet should be fine.

While the read `None` path is hit very often, the `write` path is also only hit very rarely. So you would only benefit when trying to write *multiple* fields in the extra struct using a mutable reference and that's so rare that we don't need to worry about optimizing in that regard yet.